toolchain/parse/state.def

```diff
@@ -730,8 +730,8 @@ CARBON_PARSE_STATE_VARIANTS3(TypeAfterParams, Class, Interface,
 // Always:
 //   1. PatternAsVariable
 //   2. VarAfterPattern
-//   3. VarFinishAs(Semicolon|For)
-CARBON_PARSE_STATE_VARIANTS2(Var, Semicolon, For)
+//   3. VarFinishAs(Declaration|For)
+CARBON_PARSE_STATE_VARIANTS2(Var, Declaration, For)
```

Comment:
You've changed both what this state handles and its possible outcomes; please update the documentation.

Reply:
Done.